src/baldr/edgeinfo.cc

```diff
@@ -70,17 +77,21 @@ NameInfo EdgeInfo::GetNameInfo(uint8_t index) const {
 }
 
 // Get a list of names
-std::vector<std::string> EdgeInfo::GetNames(bool only_tagged_names) const {
+std::vector<std::string> EdgeInfo::GetNames(bool only_tagged_values) const {
   // Get each name
   std::vector<std::string> names;
   names.reserve(name_count());
   const NameInfo* ni = name_info_list_;
   for (uint32_t i = 0; i < name_count(); i++, ni++) {
-    if ((only_tagged_names && !ni->tagged_) || (!only_tagged_names && ni->tagged_)) {
+    if ((only_tagged_values && !ni->tagged_) || (!only_tagged_values && ni->tagged_)) {
       continue;
     }
     if (ni->name_offset_ < names_list_length_) {
-      names.push_back(names_list_ + ni->name_offset_);
+      std::string name = names_list_ + ni->name_offset_;
+      if (ni->tagged_ && !IsNameTag(name[0])) {
+        continue;
+      }
+      names.emplace_back(std::move(name));
```

Comment:
this function returns "tagged" names with tag, is it by design? I was going to fix it in this PR, but we even have tests which check this behaviour

@kevinkreiser wdyt?

Reply:
i would say leave it unchanged in this PR. seems like callers of this function must check what tags they see. which makes me think that maybe the fix for the encoding might have broken those downstream callers of this... we should check who calls this and make an issue to fix it if it doesn't look robust to handling the first char. cc @dgearhart

Reply:
agree, also I splitted `GetNames(only_tagged_values)` into 2 separate functions `GetNames()` and `GetTaggedValues()` - seems to be less confusing.